src/cc/mod.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019, Cloudflare, Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+use std::time::Instant;
+
+use crate::cc;
+use crate::recovery::Sent;
+
+// Congestion Control constants
+pub const INITIAL_WINDOW_PACKETS: usize = 10;
+
+pub const INITIAL_WINDOW: usize = INITIAL_WINDOW_PACKETS * MAX_DATAGRAM_SIZE;
+
+pub const MINIMUM_WINDOW: usize = 2 * MAX_DATAGRAM_SIZE;
+
+pub const MAX_DATAGRAM_SIZE: usize = 1452;
+
+pub const LOSS_REDUCTION_FACTOR: f64 = 0.5;
+
+// Default congestion control algorithm
+pub const DEFAULT_CC_ALGORITHM: &str = cc::reno::CC_NAME_RENO;
+
+// Congestion Control Trait
+pub trait CongestionControlTrait {
+    fn new() -> Self
+    where
+        Self: Sized;
+
+    fn get_name(&self) -> &str;
+
+    fn to_string(&self) -> String {
+        self.get_name().to_string()
+    }
+
+    fn set_trace_id(&mut self, id: String);
+
+    // Access to internal variables
+    fn cwnd(&self) -> usize;
+
+    fn set_cwnd(&mut self, cwnd: usize);
```

Comment:
This is for setting initcwnd via config.set_cc_initcwnd() API. See initcwnd API comment below.

Reply:
So, it looks like this is now only needed because of `collapse_cwnd()`, can we just make that the method that implementations need to implement? That would simplify the API.

Reply:
Ok. set_cwnd() removed.